src/main/java/com/aws/greengrass/clientdevices/auth/session/SessionImpl.java

```diff
@@ -45,4 +45,33 @@ public DeviceAttribute getSessionAttribute(String attributeNamespace, String att
         }
         return null;
     }
+
+
+    /**
+     * Check if attribute provider exists.
+     *
+     * @param attributeProviderNameSpace Attribute provider namespace
+     * @return True/false
+     */
+    @Override
+    public boolean containsAttributeProvider(String attributeProviderNameSpace) {
+        return this.containsKey(attributeProviderNameSpace);
+    }
+
+
+    /**
+     * Check if attribute exists.
+     *
+     * @param attributeNamespace Attribute namespace
+     * @param attributeName      Attribute name
+     * @return True/false
+     */
+    @Override
+    public boolean containsSessionAttribute(String attributeNamespace, String attributeName) {
+        return containsAttributeProvider(attributeNamespace)
+                && this.getAttributeProvider(attributeNamespace)
```

Comment:
don't do a "contains" check and then do a "get". Instead just do a get and handle the null case. With this current implementation there's a race after checking contains